pkg/inject/oauth_proxy_test.go

```diff
@@ -181,6 +231,32 @@ func TestOAuthProxyOrderOfArguments(t *testing.T) {
 	assert.Equal(t, sortedArgs, dep.Spec.Template.Spec.Containers[1].Args)
 }
 
+func TestNoneOpenShiftOAuthProxyDefaultImageIsKeycloakGatekeeper(t *testing.T) {
+    jaeger := v1.NewJaeger(types.NamespacedName{Name: "my-instance"})
+    jaeger.Spec.Ingress.Security = v1.IngressSecurityOAuthProxy
+    viper.SetDefault("oauth-proxy-image", "quay.io/keycloak/keycloak-gatekeeper:10.0.0")
+    viper.Set("platform", "kubernetes")
+    defer viper.Reset()
+    dep := OAuthProxy(jaeger, deployment.NewQuery(jaeger).Get())
+
+    assert.Len(t, dep.Spec.Template.Spec.Containers, 2)
+    assert.Equal(t, "", jaeger.Spec.Query.OauthProxy.Image)
+    assert.Equal(t, "quay.io/keycloak/keycloak-gatekeeper:10.0.0", dep.Spec.Template.Spec.Containers[1].Image)
```

Comment:
It would probably be easier to maintain if you compare only the first part of the image, without the version (`quay.io/keycloak/keycloak-gatekeeper` vs. `quay.io/keycloak/keycloak-gatekeeper:10.0.0`)